test/listen.test.js

```diff
@@ -7,49 +7,90 @@ const test = require('tap').test
 const Fastify = require('..')
 
 test('listen accepts a port and a callback', t => {
-  t.plan(3)
+  t.plan(2)
   const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
   fastify.listen(0, (err) => {
     fastify.server.unref()
     t.is(fastify.server.address().address, '127.0.0.1')
     t.error(err)
-    t.pass()
-    fastify.close()
   })
 })
 
-test('listen accepts a port, address, and callback', t => {
+test('listen accepts a port and a callback with (err, address)', t => {
   t.plan(2)
   const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+  fastify.listen(0, (err, address) => {
+    fastify.server.unref()
```

Comment:
There is no need to use both `fastify.server.unref()` and `t.tearDown(fastify.close.bind(fastify))`.